### ISSUE TITLE
Enum#group_by docs incorrect re: ordering

### ISSUE THREAD

**Issue by Cohen-Carlisle** [elixir-lang/elixir#6759]
### Environment

* Elixir & Erlang versions: Erlang/OTP 19, Elixir 1.4.2
* Operating system: macOS Sierra Version 10.12.6 (16G29)

### Current behavior

```elixir
iex(1)> Enum.group_by(~w{ant buffalo cat dingo}, &String.length/1)
%{3 => ["ant", "cat"], 5 => ["dingo"], 7 => ["buffalo"]}
```

### Expected behavior

From [the docs](https://hexdocs.pm/elixir/1.3.4/Enum.html#group_by/3)
```elixir
iex> Enum.group_by(~w{ant buffalo cat dingo}, &String.length/1)
%{3 => ["ant", "cat"], 7 => ["buffalo"], 5 => ["dingo"]}
```

I'm guessing the docs are gonna change to reflect current behavior, though.

**Comment by OvermindDL1** [comment]
I'm not sure what is wrong there?  The only list shown has the same order in both of your examples?

If you are talking about the map ordering, well that does not matter, maps are orderless (well internally they are ordered until about 32 elements in size then they are not, but they should always be considered unordered).

**Comment by josevalim** [comment]
@OvermindDL1 it is regarding the map.

@Cohen-Carlisle thanks! We will change the docs but keep in mind that the order in maps absolutely does not matter, although we should keep the examples consistent to avoid confusion. :)

**Comment by Cohen-Carlisle** [comment]
Ah, I guess I misinterpreted what "Ordering is preserved." from the docs meant. Based on that phrase and the example given, I thought the map keys would be ordered by when the first element matching that key was encountered. In other words, I thought that the position of `7` as the second key was due to the fact the `7` was the second length of string encountered.

I blame ruby for making me assume that maps would be ordered 😆
